### PR TITLE
Fix parser error on forward slash

### LINKED ISSUE #407: Forward slashes result in build error when used in template strings within inline functions
Here's a Codesandbox demo of the error https://codesandbox.io/s/dreamy-darkness-dd1vi?file=/src/pages/index.astro

The following results in an error `ParseError: <a> was left open`

```
---
let tags = ["dogs", "cats", "birbs"]
---

<body>
  {tags.map(t => <a href={`/tags/${t}`}>{t}</a>)}
</body>
```

### Workaround

As long as you don't create the template string directly in the expression, you can make this work:

```
---
let tags = ["dogs", "cats", "birbs"]
---

<body>
  {tags.map(t => {
    const href = `/tags/${t}`;
    return <a href={href}>{t}</a>
  })}
</body>
```

or if you have a URL creator function somewhere:

```
---
let tags = ["dogs", "cats", "birbs"]
let createUrl = (s) => `/tags/${s}`;
---

<body>
  {tags.map(t => <a href={createUrl(t)}>{t}</a>)}
</body>
```

### PR DESCRIPTION
**Pull request by drwpow**
## Changes

Fixes #407

## Testing

Adds test

## Docs

Internal change

### CODE REVIEW

File: packages/astro-parser/src/parse/read/expression.ts [hunk 1]
```diff
@@ -84,7 +84,8 @@ function consume_tag(state: ParseState) {
 
     switch (char) {
       case "'":
-      case '"': {
+      case '"':
+      case '`': {
```
Comment (drwpow) [inline packages/astro-parser/src/parse/read/expression.ts:88]: This was the bug in the parser: we forgot to look for "`" as a valid string character.

This behavior was inconsistent with the rest of the parser which usually looks for all 3 string characters. And as far as I can tell, this deviation in this one spot seems accidental.

File: packages/astro/test/astro-basic.test.js [hunk 2]
```diff
@@ -64,4 +64,14 @@ Basics('Build does not include HMR client', async ({ build, readFile }) => {
   assert.equal(hmrPortScript.length, 0, 'No script setting the websocket port');
 });
 
+Basics('Allows forward-slashes in mustache tags (#407)', async ({ runtime }) => {
```
Comment (drwpow) [inline packages/astro/test/astro-basic.test.js:67]: Test added

File: packages/astro-parser/src/parse/read/expression.ts (hunk 1 above)
Comment (natemoo-re) [reply packages/astro-parser/src/parse/read/expression.ts:88]: Yep I think you're correct!

**Review by natemoo-re — APPROVED** [review]
LGTM! Nice catch.

File: packages/astro-parser/src/parse/read/expression.ts (hunk 1 above)
Comment (matthewp) [reply packages/astro-parser/src/parse/read/expression.ts:88]: This was me

Comment (drwpow) [reply packages/astro-parser/src/parse/read/expression.ts:88]: > This was me

I SAID **WE** MATTHEW THAT WASN’T AN ACCIDENT

**Pull request merged**